MIGRATION.md

```diff
@@ -0,0 +1,72 @@
+# Migration guide
+
+## Migration from 0.13 to 0.14
+
+If you're migrating from 0.13 it's fairly easy to do, under the right circumstances you can also serve a file that is compatible with both versions at the same time.
+
+### 0.13 and 0.14 in the same file
+If you're not using the barometer sensor or provide a `null` value to the `status.open` field just add the following, and you're good to go! 
+
+```json
+{
+  ...
+  "api_compatibility":  [ "0.14" ],
+  ...
+}
+```
+This is of course just the most basic thing, check out the [changelog](./CHANGELOG.md) for new or renamed fields too!
+
+
+### Compatibility issues
+
+#### status.open
+
+The `open` field can't have a `null` value anymore, but we've also removed the requirement.
+If you don't want to show your opening status you had to provide this (which you can now completely remove):
+```json
+{
+  ...
+  "status": {
+    "open": null
+  },
+  ...
+}
+```
+
+#### barometer
+If you're using the barometer sensor you also can't serve 0.13 anymore. 
+You have to switch the value in the unit field to `hPa`.
+
+So this:
+```json
+{
+  ...
+  "sensors":  {
+    "barometer": [
+      {
+        "value": 1020,
+        "unit": "hPA",
```

Comment:
I have been thinking about this: Can we just accept both units in 0.14 but deprecate the old one? So that you are still allowed to have the old unit *if* you also provide 0.13 compatibility?
So kind of if you want to serve only 0.14 then you *must* use the new version, but if you advertise support for both 0.14 and 0.13 you may use the old notation.

Reply:
Doesn't sound bad, right now there are 3 spaces that use the field (two of them are invalid fyi)
* i3detroit.org
* c3l.lu
* hasi.it

Is there a possibility to deprecate values in `draft-07`?

Reply:
I think that would make validation quite complex, because whether a value is valid or not depends on other values (e.g. 0.15) being absent.

Reply:
But of course we could make both "hPA" and "hPa" valid in 0.14 and simply describe "hPA" as being deprecated. Then we remove it in 0.15. We don't necessarily need a formal deprecation process for that, as long as people can support both 0.13 and 0.14 with the same sensor.

Reply:
> Is there a possibility to deprecate values in draft-07?

It's part of the current validation draft: http://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.9.3 However, the way I understand it, only entire properties can be deprecated, not enum variants.

Reply:
> > Is there a possibility to deprecate values in draft-07?
> 
> It's part of the current validation draft: http://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.9.3 However, the way I understand it, only entire properties can be deprecated, not enum variants.

Ah, damn, then it's no good.

Reply:
> But of course we could make both "hPA" and "hPa" valid in 0.14 and simply describe "hPA" as being deprecated. Then we remove it in 0.15. We don't necessarily need a formal deprecation process for that, as long as people can support both 0.13 and 0.14 with the same sensor.

Yeah, that would be the proper action i would say, checking if 0.15 is absent doesn't make any sense since 0.13 and 0.14 should ignore it, removing it in 0.15 would give you the possibility to be valid against the current and the former active version (or current and draft)

Reply:
https://github.com/SpaceApi/schema/pull/72

After that, the migration guide should probably be updated. The old unit should *only* be used for compatibility reasons.